static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -57,50 +137,102 @@ function WidgetBuilder({
     query: {...location.query, dataSet: undefined},
   };
 
-  const [title, setTitle] = useState(
-    t('Custom %s Widget', displayTypes[DisplayType.AREA])
-  );
+  // Construct PageFilters object using statsPeriod/start/end props so we can
+  // render widget graph using saved timeframe from Saved/Prebuilt Query
+  const pageFilters: PageFilters = statsPeriod
+    ? {...selection, datetime: {start: null, end: null, period: statsPeriod, utc: null}}
+    : start && end
+    ? {...selection, datetime: {start, end, period: null, utc: null}}
+    : selection;
+
+  // when opening from discover or issues page, the user selects the dashboard in the widget UI
+  const omitDashboardProp = [
+    DashboardWidgetSource.DISCOVERV2,
+    DashboardWidgetSource.ISSUE_DETAILS,
+  ].includes(source);
 
-  useEffect(() => {
-    if (!dataSet) {
-      handleDataSetChange(DataSet.EVENTS);
+  const [state, setState] = useState<State>(() => {
+    if (!widget) {
+      return {
+        title:
+          defaultTitle ??
+          t('Custom %s Widget', displayTypes[displayType ?? DisplayType.TABLE]),
+        displayType: displayType ?? DisplayType.TABLE,
+        interval: '5m',
+        queries: [defaultWidgetQuery ? {...defaultWidgetQuery} : {...QUERIES.events}],
+        errors: undefined,
+        loading: !!omitDashboardProp,
```

Comment:
Related to my comment about `omitDashboardProp`, I wonder if we really need to keep this loading variable in state.

Reply:
please lets keep it for now and see how the widget builder will work when opened from other verticals